File: VintedProject/src/main/java/it/unical/ea/VintedProject/dto/OrderDto.java
package it.unical.ea.VintedProject.dto;
import it.unical.ea.VintedProject.data.entities.BasicInsertion;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;


import java.time.LocalDate;
import java.util.List;


@Data
@NoArgsConstructor
@ToString
public class OrderDto {

    @NotNull
    private Long id;

    private LocalDate date;

    @NotNull
    private Long paymentId;

    private List<Long> insertionIdList;

    @NotNull
    private Long userId;

}

File: VintedAndroid/app/src/main/java/com/example/vintedandroid/dto/OrderDto.kt
package com.example.vintedandroid.dto

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate

@Entity(tableName = "orderDto")
class OrderDto (
    @PrimaryKey val id : Long,
    @ColumnInfo val date: LocalDate,
    @ColumnInfo val paymentid : Long,
    @ColumnInfo val number : Int,
    @ColumnInfo val inserionId : Long,
    @ColumnInfo val userId : Long
        ){
}